astroid/rebuilder.py

```diff
@@ -113,12 +114,16 @@ def _get_doc(self, node: T_Doc) -> Tuple[T_Doc, Optional[str]]:
                     and isinstance(first_value, self._module.Constant)
                     and isinstance(first_value.value, str)
                 ):
-                    doc = first_value.value if PY38_PLUS else first_value.s
+                    doc_node = self.visit(node.body[0], node)
```

Comment:
This won't work unfortunately. `node` here is still an `ast` node. However the `self.visit` method needs an astroid node which can be assigned as `parent` later.

I.e. the `doc_node` can't be build here. Instead just return the `node.body[0]` ast node for now. You can keep the `col_offset` handling, by doing it on the ast node directly.

Reply:
I get what you mean but not sure how teh code should look like. Can you please assist

Reply:
Here, inside the if block, it should be something like this
```py
                    doc = first_value.value if PY38_PLUS else first_value.s
                    doc_node = node.body[0]
                    node.body = node.body[1:]
                    # The ast parser of python < 3.8 sets col_offset of multi-line strings to -1
                    # as it is unable to determine the value correctly. We reset this to None.
                    if doc_node.col_offset == -1:
                        doc_node.col_offset = None
                    return doc, doc_node
```
Note that `doc_node` is just `node.body[0]` instead of the return from `self.visit`.
For the other changes, see the next comments.